xkafka/consumer.go

```diff
@@ -0,0 +1,154 @@
+package xkafka
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/sourcegraph/conc/stream"
+)
+
+// KafkaConsumer is the interface for kafka consumer.
+type KafkaConsumer interface {
```

Comment:
Can we break this into smaller interfaces? Fatty interfaces are usually not a good idea and harder to use.

Reply:
This is an interface for https://github.com/confluentinc/confluent-kafka-go/blob/master/kafka/consumer.go#L43 Using this interface to inject mock for testing

Can maybe rename/move to avoid confusion with `xkafka.Consumer`

Reply:
Can you make it private interface then? We should be mindful of the public API, because it is hard to change it later on.

Reply:
Moved it to `internal` package. This will allow exposing the option, while keeping the interface private